### PR TITLE
LPS-135621 put redirect parameter to the parent folder of the file to unify the behavior if the cases notification or mail

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-135621

The link to mails and inline notification (or other methods) is the same. We can not be sure if the "back"/"publish" is to the notifications page or a new page or is from different user, so I think is better to unify the behavior, that the redirect is to the DM (folder) of the file.

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — COMMENTED** [review]
@AliciaGarciaGarcia Why all this renaming and reordering? Why don't we just keep the same order for the old parameters, and simply add the redirect at the end? All this changes make reviewing this a bit confusing; if you need to resort, reorganize the PR in two commits, one adding the changes, and another one for the sort/rename.

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa

thanks!

I remade it a little, see if now is more tidy :)

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2729,13 +2729,16 @@ private String _buildEntryURL(DLFileVersion fileVersion)
...
 			fileVersion.getGroupId(), portletId, null);
 
 		String namespace = PortalUtil.getPortletNamespace(portletId);
 
-		entryURL = HttpUtil.addParameter(
-			entryURL, namespace + "mvcRenderCommandName",
+		redirectURL = HttpUtil.addParameter(
+			redirectURL, namespace + "folderId", fileVersion.getFolderId());
+
+		String entryURL = HttpUtil.addParameter(
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2740]: This reuse of the redirectURL is a bit confusing. Even if both the entry and redirect URL share some common parameters, I think it is better to keep the separate. Can you create a new private "_getRedirectURL()" method and use that in _buildEntryURL?

Even if there's a bit of duplication, I think it is much clearer to keep them separated.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java]: Perfect!

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**